ClientFactory/AutoDiscoveryFactory.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace Http\HttplugBundle\ClientFactory;
+
+use Http\Discovery\HttpClientDiscovery;
+
+/**
+ * Use auto discovery to find a HTTP client.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class AutoDiscoveryFactory implements ClientFactory
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function createClient(array $config = [])
+    {
+        return HttpClientDiscovery::find();
```

Comment:
I doubt it. I believe it is very rare that an async adapter/client does not implement HttpClient interface (not async).